containers/src/Data/Map/Internal.hs

```diff
@@ -2085,6 +2088,25 @@ disjoint (Bin _ k _ l r) t
   where
     (lt,found,gt) = splitMember k t
 
+{--------------------------------------------------------------------
+  Compose
+--------------------------------------------------------------------}
+-- | /O(|ab|*log(|bc|))/. Relate the keys of one map to the values of
```

Comment:
This bound is a bit funny when |bc| ≤ 1. Is it worth fixing? Not sure.

Reply:
Wouldn't that also apply to the existing `O(n log n)` bounds?
Big O notation is asymptotic, so in a sense there's nothing to fix.
Is there something specific about this bound you intend to make clearer? e.g. the `if null bc` case